src/de.rs

```diff
@@ -959,6 +976,71 @@ mod test {
 		);
 	}
 
+	#[test]
+	fn test_payment_secret_deserialization() {
+		use bech32::u5;
+		use secp256k1::recovery::{RecoveryId, RecoverableSignature};
+		use TaggedField::*;
+		use {SiPrefix, SignedRawInvoice, Signature, RawInvoice, RawTaggedField, RawHrp, RawDataPart,
+				 Currency, Sha256, PositiveTimestamp};
+
+		assert_eq!( // BOLT 11 payment secret invoice. The unknown fields are invoice features.
+			"lnbc25m1pvjluezpp5qqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypqdq5vdhkven9v5sxyetpdeessp5zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zyg3zygs9q5sqqqqqqqqqqqqqqqpqsq67gye39hfg3zd8rgc80k32tvy9xk2xunwm5lzexnvpx6fd77en8qaq424dxgt56cag2dpt359k3ssyhetktkpqh24jqnjyw6uqd08sgptq44qu".parse(),
+			Ok(SignedRawInvoice {
+					raw_invoice: RawInvoice {
+						hrp: RawHrp {
+							currency: Currency::Bitcoin,
+							raw_amount: Some(25),
+							si_prefix: Some(SiPrefix::Milli)
+						},
+						data: RawDataPart {
+							timestamp: PositiveTimestamp::from_unix_timestamp(1496314658).unwrap(),
+							tagged_fields: vec ! [
+								PaymentHash(Sha256(sha256::Hash::from_hex(
+									"0001020304050607080900010203040506070809000102030405060708090102"
+								).unwrap())).into(),
+								Description(::Description::new("coffee beans".to_owned()).unwrap()).into(),
+								PaymentSecret(::PaymentSecret([17; 32])).into(),
+								RawTaggedField::UnknownSemantics(vec![u5::try_from_u8(5).unwrap(),
+																											u5::try_from_u8(0).unwrap(),
```

Comment:
Sorry for this nit, but this really isn't elegant. We should be able to construct `u5`s directly here. There is also https://docs.rs/bech32/0.8.0/bech32/trait.CheckBase32.html#impl-CheckBase32%3CVec%3Cu5%2C%20Global%3E%3E to convert a `u8` slice to a `Vec<u5>`.

Reply:
Ah no, that's waaay more elegant, thanks :) updated.